recipes/compute_library/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ARM-software/ComputeLibrary"
+    topics = ("android", "linux", "machine-learning", "arm", "computer-vision", "neural-network")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "enable_openmp": [True, False],
+        "enable_opencl": [True, False],
+        "enable_neon": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "enable_openmp": False,
+        "enable_opencl": True,
+        "enable_neon": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 14
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "10",
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def config_options(self):
+        # INFO: Neon option is reserved to arm architecture
+        if "arm" not in str(self.settings.arch):
+            del self.options.enable_neon
+        # INFO: OpenMP option only works with g++, according to the documentation
+        if self.settings.compiler == "clang":
+            del self.options.enable_openmp
+        # INFO: OpenCL fails to build with MacOS
+        if self.settings.os == "Macos":
+            self.options.enable_opencl = False
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("scons/4.3.0")
+
+    def requirements(self):
+        if self.options.enable_opencl:
+            self.requires("opencl-headers/2023.04.17")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.")
+        supported_os = ["Android", "Linux", "OpenBSD", "Macos", "Tizen"]
+        if str(self.settings.os) not in supported_os:
+            raise ConanInvalidConfiguration(f"{self.ref} does not support {self.settings.os}. It is only supported on {supported_os}.")
+        if "arm" not in str(self.settings.arch) and "x86" not in str(self.settings.arch):
+            raise ConanInvalidConfiguration(f"{self.ref} does not support {self.settings.arch}. It is only supported on arm and x86.")
+        if "x86" in str(self.settings.arch) and (self.options.get_safe("enable_neon") or not self.options.enable_opencl):
```

Comment:
NEON option is already removed in case of non-ARM platform, so do we need to validate it here?